### ISSUE TITLE
Clustering task fails to upload

### ISSUE THREAD

**Issue by ArlindKadra** [openml/OpenML#925]
One of the unit tests that I added for uploading clustering tasks is failing. 
This is the task xml:
```
<oml:task_inputs xmlns:oml="http://openml.org/openml">
	<oml:task_type_id>5</oml:task_type_id>
	<oml:input name="source_data">77</oml:input>
</oml:task_inputs>
```
It fails with the following error:
`openml.exceptions.OpenMLServerException: https://test.openml.org/api/v1/xml/task/ returned code 618: Task does not contain all required inputs.`

I verified with the following xsd and it is validated:
https://github.com/openml/OpenML/blob/develop/openml_OS/views/pages/api_new/v1/xsd/openml.task.upload.xsd

One more question is why is target_feature required in clustering tasks:
https://www.openml.org/tt/5
It is empty in the clustering tasks that I checked in OpenML and when it is not on the clustering task class in the openml-python  connector:
https://github.com/openml/openml-python/blob/96db525e650e527b3283fb64bace8509639ce1f4/openml/tasks/task.py#L167

**Comment by janvanrijn** [comment]
(Please ping me with these kind of issues, I totally missed this one.)

Before creating a task, please have a look at the task type. 
https://www.openml.org/api/v1/tasktype/<ttid>

For clustering, it is the following:
https://www.openml.org/api/v1/tasktype/5

It fails because it requires a target feature as well. 
Not sure why it has a target feature, is this conventional? @joaquinvanschoren 

HTH

**Comment by ArlindKadra** [comment]
@janvanrijn I did. That is why I asked why is target_feature required when it is not present in the OpenML tasks and openml-python connector.

>One more question is why is target_feature required in clustering tasks:
>https://www.openml.org/tt/5
>It is empty in the clustering tasks that I checked in OpenML and it is not on the clustering task class in the >openml-python connector:
>https://github.com/openml/openml-python/blob/96db525e650e527b3283fb64bace8509639ce1f4/openml>/tasks/task.py#L167

**Comment by joaquinvanschoren** [comment]
The target feature should be optional. It is only needed when there is a 'ground truth' about which are the right clusters. This is only done for benchmarking clustering algorithms, in normal situations there is no ground truth.

**Comment by janvanrijn** [comment]
In that case we should redefine the cluster task.

**Comment by ArlindKadra** [comment]
@janvanrijn Not sure if you have already started on redefining the cluster task, but I see some discrepancies in the requirements for the clustering tasks at the test server. In there, `estimation_procedure` is listed as required and it is not present in the python ClusteringTask class.

https://test.openml.org/api/v1/tasktype/5

**Comment by janvanrijn** [comment]
Good point, thanks for pointing this out. Can you please check again? I just added it in the database.

**Comment by ArlindKadra** [comment]
`b'<oml:error xmlns:oml="http://openml.org/openml">\n\t<oml:code>618</oml:code>\n\t<oml:message>Task does not contain all required inputs.</oml:message>\n\t\t<oml:additional_information>problematic input(s): estimation_procedure</oml:additional_information>\n\t</oml:error>\n'`

**Comment by ArlindKadra** [comment]
@joaquinvanschoren @janvanrijn I am noticing the requirements for the upload of clustering tasks and evaluation measures is required. I am guessing this is to be used in conjunction with the target feature 'ground truth'. Since target feature should be optional, shouldn't evaluation measure be optional too ?

**Comment by janvanrijn** [comment]
@ArlindKadra I got an email notification on this thread, yet I can not find the question here. Did something change?

**Comment by ArlindKadra** [comment]
@janvanrijn The notification is right, I posted a question, but then I deleted it since it was not related to the OpenML repo. Basically we changed evaluation_measure in clustering tasks to not be mandatory in accordance with the target feature.
